modules/custom/az_paragraphs/az_paragraphs_text_media/js/az_paragraphs_az_text_media_vimeo.es6.js

```diff
@@ -0,0 +1,189 @@
+
+            // Play Button
+            const playButton =
+              bgVideoParagraphs[index].getElementsByClassName(
+                'az-video-play',
+              )[0];
+            playButton.addEventListener('click', async (event) => {
+              event.preventDefault();
+              bgVideoParagraphs[index].player
+                .play()
+                .then(() => {
+                  // the video is playing
+                })
+                .catch((error) => {
+                  switch (error.name) {
+                    case 'PasswordError':
+                      window.alert('the video is password-protected');
+                      break;
+                    case 'PrivacyError':
+                      window.alert('the video is private');
+                      break;
+                    default:
+                      console.log(`Some errors occurred: ${error.name}`);
+                      break;
+                  }
+                });
+            });
+
+            // Pause Button
+            const pauseButton =
+              bgVideoParagraphs[index].getElementsByClassName(
+                'az-video-pause',
+              )[0];
+            pauseButton.addEventListener('click', async (event) => {
+              event.preventDefault();
+              bgVideoParagraphs[index].player
+                .pause()
+                .then(() => {
+                  // the video is paused
+                })
+                .catch((error) => {
+                  switch (error.name) {
+                    case 'PasswordError':
+                      window.alert('the video is password-protected');
+                      break;
+                    case 'PrivacyError':
+                      window.alert('the video is private');
+                      break;
+                    default:
+                      window.alert(`Some errors occurred: ${error.name}`);
+                      break;
+                  }
+                });
+            });
+          });
+        };
+
+        // Resize Logic
+        const setDimensions = (container) => {
+          const parentParagraph = container.parentNode;
+          const vimeoId = container.dataset.vimeoId2;
+          const thisPlayer =
+            container.getElementsByClassName('az-video-player')[0].firstChild;
+          thisPlayer.style.zIndex = -100;
+          const { style } = container.dataset;
+          const width = container.offsetWidth;
+          const height = container.offsetHeight;
+          const { ratio } = bgVideos[vimeoId];
+          const pWidth = Math.ceil(height * ratio); // get new player width
+          const pHeight = Math.ceil(width / ratio); // get new player height
+          let parentHeight = parentParagraph.offsetHeight;
+          parentHeight = `${parentHeight.toString()}px`;
+          container.style.height = parentHeight;
+          if (style === 'bottom') {
+            container.style.top = 0;
+          }
```

Comment:
I see a console error sometimes since thisPlayer can be null (before the iframe is initialized). I added an if statement for that case and reorganized the styling changes so that we can apply the changes to the parent and then stop if the player is null.

```suggestion
          let parentHeight = parentParagraph.offsetHeight;
          parentHeight = `${parentHeight.toString()}px`;
          container.style.height = parentHeight;
          const { style } = container.dataset;
          if (style === 'bottom') {
            container.style.top = 0;
          }
          const thisPlayer =
            container.getElementsByClassName('az-video-player')[0].firstChild;
          if (thisPlayer === null) {
            return;
          }
          thisPlayer.style.zIndex = -100;
          const vimeoId = container.dataset.vimeoId2;
          const width = container.offsetWidth;
          const height = container.offsetHeight;
          const { ratio } = bgVideos[vimeoId];
          const pWidth = Math.ceil(height * ratio); // get new player width
          const pHeight = Math.ceil(width / ratio); // get new player height
```